plugin/pkg/admission/webhook/admission.go

```diff
@@ -265,27 +256,10 @@ func (a *GenericAdmissionWebhook) hookClient(h *v1alpha1.ExternalAdmissionHook)
 		return nil, err
 	}
 
-	var dial func(network, addr string) (net.Conn, error)
-	if a.proxyTransport != nil && a.proxyTransport.Dial != nil {
-		dial = a.proxyTransport.Dial
-	}
-
 	// TODO: cache these instead of constructing one each time
-	cfg := &rest.Config{
-		Host:    u.Host,
-		APIPath: u.Path,
-		TLSClientConfig: rest.TLSClientConfig{
-			ServerName: h.ClientConfig.Service.Name + "." + h.ClientConfig.Service.Namespace + ".svc",
-			CAData:     h.ClientConfig.CABundle,
-			CertData:   a.clientCert,
-			KeyData:    a.clientKey,
-		},
-		UserAgent: "kube-apiserver-admission",
-		Timeout:   30 * time.Second,
-		ContentConfig: rest.ContentConfig{
-			NegotiatedSerializer: a.negotiatedSerializer,
-		},
-		Dial: dial,
-	}
+	cfg := rest.CopyConfig(a.restClientConfig)
```

Comment:
useragent and timeout still seem relevant

Reply:
> useragent and timeout still seem relevant

Want to do it here or in the client being used to initialize?  I don't feel strongly.

Reply:
passing in the values is ok (probably better than stomping here). I think we definitely want the timeout

Reply:
Where do you set the `ServerName` and `CABundle`?

Reply:
> Where do you set the ServerName and CABundle?

missed them.  Put them just below.